sympy/core/numbers.py

```diff
@@ -3413,6 +3416,216 @@ def __hash__(self):
         return super(NumberSymbol, self).__hash__()
 
 
+class BigNumber:
```

Comment:
I think that it's just better to inherit `Basic` or `Expr` if you expect this to work with sympy operators.

Reply:
ok, that makes sense to me. I'll inherit from Expr, I think. Or maybe unevaluated expression?